src/deploy.ts

```diff
@@ -78,7 +79,7 @@ type DeployTargetInfo =
 
 /** Deploy a project to ObservableHQ */
 export async function deploy(
-  {config, message, force, deployPollInterval = DEPLOY_POLL_INTERVAL_MS, maxConcurrency}: DeployOptions,
+  {config, message, force, deployPollInterval = DEPLOY_POLL_INTERVAL_MS, maxConcurrency, deployId}: DeployOptions,
```

Comment:
> but if a project doesn't have a deploy.json file, we should still be able to deploy with --id, without prompting for the project.

Could we read the deploy config in observable.ts and pass it down to deploy.deploy()? And if we have a deployId, override the config with a stub config containing only the id?

Reply:
Check out my updated handling of deploy id provided vs. not.